### ISSUE TITLE
macOS Helper apps without signing do not have sandbox entitlements, but with the main app crashes

### ISSUE THREAD

**Issue by stephendwolff** [electron/electron-packager#1054]
### Issue Details

* **Electron Packager Version:**
  * 14.0.6 and master

* **Electron Version:**
  * 6.0.11

* **Operating System:**
  * macOS

### Expected Behavior
An app built using electron-packager, should run and have sandbox entitlements for Helper Apps (GPU, Renderer and Plugin) for submission to the mac app store (mas)

### Actual Behavior
An app built using electron-packager runs when signed following the electron [mac app store instructions](https://github.com/electron/electron/blob/master/docs/tutorial/mac-app-store-submission-guide.md), but when uploaded using Application Loader, there is an error saying that the Helper Apps do not have sandbox entitlements.

If the script is modified to add sandbox entitlements and sign the Helper Apps (GPU, Plugin, Renderer), the app no longer runs, but crashes with 'Crashed Thread:        24  ThreadPoolSingleThreadForegroundBlocking0......'

### To Reproduce
My original packager command is:

```
electron-packager . --app-bundle-id=com.weareamplify.disconnect --app-category-type=public.app-category.lifestyle --overwrite --platform=mas --arch=x64 --icon=Icon.icns --prune=true --out=mas-builds
```

My signing script is:
```
#!/bin/bash

# Name of your app.
APP="disconnect"
APP_FOLDER="/Users/stephen/Documents/Development/amplify/disconnected/mas-builds/disconnect-mas-x64"
# The path of your app to sign.
APP_PATH="$APP_FOLDER/disconnect.app"
# The path to the location you want to put the signed package.
RESULT_PATH="$APP_FOLDER/$APP.pkg"
# The name of certificates you requested.
APP_KEY="3rd Party Mac Developer Application: Louder Productions Limited (HKH4LU2D76)"
INSTALLER_KEY="3rd Party Mac Developer Installer: Louder Productions Limited (HKH4LU2D76)"
# The path of your plist files.
CHILD_PLIST="/Users/stephen/Documents/Development/amplify/disconnected/child.plist"
PARENT_PLIST="/Users/stephen/Documents/Development/amplify/disconnected/parent.plist"
LOGINHELPER_PLIST="/Users/stephen/Documents/Development/amplify/disconnected/loginhelper.plist"

FRAMEWORKS_PATH="$APP_PATH/Contents/Frameworks"

codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/Electron Framework.framework/Versions/A/Electron Framework"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/Electron Framework.framework/Versions/A/Libraries/libffmpeg.dylib"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/Electron Framework.framework"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper.app/Contents/MacOS/$APP Helper"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper (GPU).app/Contents/MacOS/$APP Helper (GPU)"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper (Plugin).app/Contents/MacOS/$APP Helper (Plugin)"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper (Renderer).app/Contents/MacOS/$APP Helper (Renderer)"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper.app/"
codesign -s "$APP_KEY" -f --entitlements "$LOGINHELPER_PLIST" "$APP_PATH/Contents/Library/LoginItems/$APP Login Helper.app/Contents/MacOS/$APP Login Helper"
codesign -s "$APP_KEY" -f --entitlements "$LOGINHELPER_PLIST" "$APP_PATH/Contents/Library/LoginItems/$APP Login Helper.app/"
codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$APP_PATH/Contents/MacOS/$APP"
codesign -s "$APP_KEY" -f --entitlements "$PARENT_PLIST" "$APP_PATH"

productbuild --component "$APP_PATH" /Applications --sign "$INSTALLER_KEY" "$RESULT_PATH"
```

### Additional Information
I have tried submitting this as a [documentation bug on the electron repository](https://github.com/electron/electron/issues/20405), but i noticed that a recent fix was made for these Helper apps in this repository, so wondered if you might know what entitlements these Helper apps might need or if there is any other reason for a crash?

Also, I note that although i can see a fix for Helper acts which is supposed to be in electron-packager 14.0.6, installing via npm does not include this fix, so i tried installing master directly.

**Comment by malept** [comment]
This isn't a bug/question related to Electron Packager itself, but rather about macOS code signing, which Packager delegates to `electron-osx-sign`. However, I discourage you from filing yet another issue in that repository - the `electron-osx-sign` maintainer was pinged in the Electron issue, so another copy of that issue would be just noise at that point.

**Comment by stephendwolff** [comment]
Thanks for looking into it for me. I think i may need to look at what these Electron Helper apps are for, and what 'entitlements' they need on a mac with its newer security features.

**Comment by stephendwolff** [comment]
@malept, i don't expect you to re-open this issue, but i think you may have missed my point about the current release on npm not including fixes that I expected: "Also, I note that although i can see a fix for Helper acts which is supposed to be in electron-packager 14.0.6, installing via npm does not include this fix, so i tried installing master directly."

**Comment by malept** [comment]
I've verified that the fixes for the Electron 6 helpers in #1036 and #1046 are in the NPM release of 14.0.6. `master` should only contain an unreleased feature unrelated to the new helpers at the time of this comment.

**Comment by stephendwolff** [comment]
ok - thanks - i must have misunderstood which fixes were meant to be in the release.

btw - i'm making progress with the Helper app entitlements, having delved into the Chromium commits...
